lib/application.js

```diff
@@ -618,6 +618,29 @@ app.listen = function listen() {
   return server.listen.apply(server, arguments);
 };
 
+/**
+ * Creates a router that inherits options from app.settings
+ * and allows them to be overriden by passing in an options
+ * object
+ *
+ * @param {Object} options
+ * @return {router}
+ * @public
+ */
+app.createRouter = function (options) {
+  options = options || {}
```

Comment:
Since this modifies the `options` below, can you make it not modify the incoming options object?

```
var opts = options || {};

// ...
opts.strict = this.enabled('strict routing')
```

Reply:
Done!

Reply:
It still looks the same. Did you commit the fix?

Reply:
@dougwilson Yeah, it just doesn't seem like it. Look at the whole file and you'll see its fine.

Reply:
@dougwilson Um yes... But it's not modifying the original object anymore because the code changed below.

Reply:
Well I mean, technically it is modifying it but only if it reverts a falsy value in which case the user didn't actually pass anything into the function and in which case it doesn't matter.

Reply:
The argument is still being redefined. Just use the code Wes suggested. Do not reassign arguments.

Reply:
It does matter due to how v8 optimized and compiles functions.

Reply:
@dougwilson Oh I didn't realize. I fixed it now.